src/transformers/pipelines/text_generation.py

```diff
@@ -105,17 +105,8 @@ def _sanitize_parameters(
             prefix_inputs = self.tokenizer(
                 prefix, padding=False, add_special_tokens=False, return_tensors=self.framework
             )
-            prefix_length = prefix_inputs["input_ids"].shape[-1]
+            generate_kwargs["prefix_length"] = prefix_inputs["input_ids"].shape[-1]
```

Comment:
Instead of applying related changes, it stores this value for downstream consumption.